cmd/init.go

```diff
@@ -19,304 +19,243 @@ import (
-	cmd.Flags().StringVarP(&p.user.keyPath, "user-key", "", "", "user keypath (default generated)")
-	cmd.Flags().BoolVarP(&p.user.create, "create-user", "", true, "create an user")
-
-	_ = cmd.Flags().MarkHidden("create-account")
-	_ = cmd.Flags().MarkHidden("create-operator")
-	_ = cmd.Flags().MarkHidden("create-user")
-
+	cmd.Flags().StringVarP(&params.Name, "name", "n", "Test", "name used for the operator, account and user")
+	cmd.Flags().StringVarP(&params.Deploy, "url", "u", "", "managed operator deploy url")
 	return cmd
 }
 
 func init() {
-	GetRootCmd().AddCommand(CreateInitCmd())
+	GetRootCmd().AddCommand(createInitCmd())
 }
 
-type InitParams struct {
-	storeRoot       string
-	environmentName string
-	operator        Entity
-	account         Entity
-	user            Entity
+type EasyCmdParams struct {
+	Name      string
+	Operator  keys
+	Account   keys
+	User      keys
+	Deploy    string
+	DeployURL string
 }
 
-func (p *InitParams) Containers() []*Entity {
-	return []*Entity{&p.operator, &p.account, &p.user}
+type keys struct {
+	KP        nkeys.KeyPair
+	PubKey    string
+	KeyPath   string
+	CredsPath string
 }
 
-func (p *InitParams) SetDefaults() error {
-	p.operator.kind = nkeys.PrefixByteOperator
-	p.account.kind = nkeys.PrefixByteAccount
-	p.user.kind = nkeys.PrefixByteUser
+func (p *EasyCmdParams) SetDefaults(ctx ActionCtx) error {
+	return nil
+}
+func (p *EasyCmdParams) PreInteractive(ctx ActionCtx) error {
+	return nil
+}
+func (p *EasyCmdParams) Load(ctx ActionCtx) error {
+	return nil
+}
 
-	var err error
-	p.storeRoot, err = filepath.Abs(GetConfig().StoreRoot)
+func (p *EasyCmdParams) PostInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *EasyCmdParams) Validate(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *EasyCmdParams) createStore() error {
+	root, err := filepath.Abs(GetConfig().StoreRoot)
 	if err != nil {
-		return fmt.Errorf("error calculating the absolute filepath for %q: %v", p.storeRoot, err)
+		return err
 	}
-
-	// if defaults are set we are not prompting
-	if p.environmentName == "" {
-		p.environmentName = "test"
+	p.Operator.KP, err = nkeys.CreateOperator()
+	if err != nil {
+		return err
+	}
+	nk := &store.NamedKey{Name: p.Name, KP: p.Operator.KP}
+	_, err = store.CreateStore(p.Name, root, nk)
+	if err != nil {
+		return err
 	}
-	if !InteractiveFlag {
-		p.SetDefaultNames()
+	GetConfig().Operator = p.Name
+	if err := GetConfig().Save(); err != nil {
+		return err
 	}
 	return nil
 }
 
-func (p *InitParams) SetDefaultNames() {
-	// set the operator name
-	if p.operator.name == "" {
-		p.operator.name = fmt.Sprintf("%s_operator", p.environmentName)
+func (p *EasyCmdParams) setOperatorDefaults(ctx ActionCtx) error {
+	oc, err := ctx.StoreCtx().Store.ReadOperatorClaim()
+	if err != nil {
+		return err
 	}
-
-	if p.account.name == "" {
-		p.account.name = fmt.Sprintf("%s_account", p.environmentName)
+	oc.OperatorServiceURLs.Add("nats://localhost:4222")
```

Comment:
should we tell them we did this somwehere?

Reply:
this is just a default value so that the:

nsc generate config ... 
stuff allows them to use the tools

Reply:
they need to change it to their env whatever that is.